enhancements/sig-architecture/224-event-based-manifestwork/README.md

```diff
@@ -0,0 +1,788 @@
+# Event Based Manifestwork
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal optimizes scaling and performance by treating manifestworks as event messages, reducing excessive storage on the hub cluster. This allows managed cluster agents to subscribe to relevant manifestworks, eliminating the need for extensive hub storage. The proposed changes also enable diverse architectures to consume ManifestWorks, while providing a client interface for seamless adoption of the event-based mechanism.
+
```

Comment:
You know this - we really do not save `hub storage`. We divert the storage from apiserver/etcd to other horizontally scalable systems. They still need to be stored. Should we clarify that.

Reply:
we do not care about where the data is stored. the proposal enables to get the message from event streaming for the work agent. As you said, the data can be on kube api server. it can be in another rest api.

Reply:
This proposal talks about source. And source generates events to which the work agent reacts. And the first proposed User Story is related to source.
Now source, as you will agree cannot just send event w/o storing it. In pure kube world, it was getting stored in etcd. In this case where will it be stored. It should not fire an event w/o ever knowing what it fired. Or when the status feedback comes, it will have to store somewhere.
So, we should care about where it is stored. Am I missing something?

Reply:
We mainly cover two kinds of sources. one is hub controller, another is restful service. For restful service, it is usually backend by a database for storage. For hub controller, the resources are  often high-level resources above ManifestWork, for example, ManifestWorkReplicateSet or Policy(we will make some change to policy controller to use manifesowork API for policy propagation and status report). These high-level resources are abstract of ManifestWork API, will be stored in ETCD.

Reply:
@morvencao totally agreed. So do you think changing this -  `eliminating the need for extensive hub storage.` to `reducing the data being stored in ETCD of hub.` will work?

Reply:
yes, updated this. thanks @bjoydeep

Reply:
looks good to me